ProcessLib/UniformDirichletBoundaryCondition.h

```diff
@@ -48,13 +49,27 @@ class UniformDirichletBoundaryCondition
     /// and the corresponding values.
     /// The ids are appended to the global_ids and the values are filled with
     /// the constant _value.
-    void initialize(MeshGeoToolsLib::MeshNodeSearcher& searcher,
-            std::vector<std::size_t>& global_ids, std::vector<double>& values)
+    void initialize(
+            MeshGeoToolsLib::MeshNodeSearcher& searcher,
+            AssemblerLib::LocalToGlobalIndexMap const& dof_table,
+            std::size_t component_id,
+            std::vector<std::size_t>& global_ids,
+            std::vector<double>& values)
     {
         // Find nodes' ids on the given mesh on which this boundary condition
         // is defined.
         std::vector<std::size_t> ids = searcher.getMeshNodeIDs(*_geometry);
 
+        // convert mesh node ids to global index for the given component
+        for (auto& id : ids)
+        {
+            MeshLib::Location l(searcher.getMesh().getID(),
```

Comment:
I would replace the `searcher.getMesh().getID()` by `searcher.getMeshId()` just forwarding that id and not exposing the mesh of the searcher. If I didn't miss something, this is the only place where node searcher's `getMesh()` is used.